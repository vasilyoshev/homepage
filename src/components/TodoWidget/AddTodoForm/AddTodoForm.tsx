import React, { useState, ChangeEvent, FormEvent } from 'react';
import { AddTodoFormProps } from 'interfaces';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import FormGroup from '@material-ui/core/FormGroup';

export const AddTodoForm: React.FC<AddTodoFormProps> = ({ addTodo }) => {
  const [newTodo, setNewTodo] = useState<string>('');

  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    setNewTodo(e.target.value);
  };

  const submitTodoItem = (e: FormEvent<HTMLButtonElement>) => {
    e.preventDefault();
    addTodo(newTodo);
    setNewTodo('');
  };

  return (
    <FormGroup>
      <TextField
        value={newTodo}
        color="primary"
        margin='normal'
        variant="outlined"
        size="small"
        onChange={handleInputChange}
      />
      <Button
        size="medium"
        variant="contained"
        color="primary"
        type="submit"
        onClick={submitTodoItem}
      >
        Add Todo
      </Button>
    </FormGroup>
  );
};
