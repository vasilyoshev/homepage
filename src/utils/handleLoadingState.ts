import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
