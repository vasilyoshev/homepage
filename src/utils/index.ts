export * from './handleLoadingState';
export * from './testUtils';
