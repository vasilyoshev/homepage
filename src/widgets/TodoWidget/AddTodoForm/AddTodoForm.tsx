import React, { useState, ChangeEvent, FormEvent } from 'react';
import { AddTodoFormProps } from 'interfaces';

export const AddTodoForm: React.FC<AddTodoFormProps> = ({ addTodo }) => {
  const [newTodo, setNewTodo] = useState<string>('');

  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    setNewTodo(e.target.value);
  };

  const submitTodoItem = (e: FormEvent<HTMLButtonElement>) => {
    e.preventDefault();
    addTodo(newTodo);
    setNewTodo('');
  };

  return (
    <form>
      <input 
        type="text" 
        value={newTodo} 
        onChange={handleInputChange} />
      <button
        type="submit" 
        onClick={submitTodoItem}>
        Add Todo
      </button>
    </form>
  );
};



