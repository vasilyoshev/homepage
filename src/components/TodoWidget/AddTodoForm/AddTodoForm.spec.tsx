import { render } from 'utils';
import { AddTodoForm } from './AddTodoForm';

describe('AddTodoForm', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<AddTodoForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
