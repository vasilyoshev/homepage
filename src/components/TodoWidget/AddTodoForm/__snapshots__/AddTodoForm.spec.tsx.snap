// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodoForm renders successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiFormGroup-root"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-2"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Add Todo
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
